bindings/python/htcondor2/_submit.py

```diff
@@ -249,22 +249,32 @@ def getQArgs(self) -> str:
 
     def setQArgs(self, args : str):
         '''
-        Set the queue statement.  This statement replaces the queue statement,
-        if any, passed to the original constructor.
+        Set the arguments to the queue statement.  These arguments replace
+        the arguments, if any, passed to the original constructor.
 
-        :param args:  The complete queue statement.
+        :param args:  The queue arguments.
         '''
         if not isinstance(args, str):
             raise TypeError("args must be a string")
         _submit_setqargs(self, self._handle, args)
 
 
-    def itemdata(self) -> Union[ Iterator[str], Iterator[dict] ]:
+    def itemdata(self, qargs : str = None) -> Union[ Iterator[str], Iterator[dict] ]:
         '''
-        Returns an iterator over the itemdata specified by the queue statement,
+        Returns an iterator over the itemdata specified by the given
+        arguments to a queue statement,
         suitable for passing to :meth:`schedd.Submit`.
+
+        ``s.itemdata()`` is equivalent to ``s.itemdata(s.getQAargs())``.
+
+        :param qargs:  A set of arguments for a queue statement.
         '''
-        (keys_str, values_str) = _submit_itemdata(self, self._handle)
+        s = self
+        if qargs is not None:
+            s = Submit()
+            s.setQArgs(qargs)
```

Comment:
if passed-in args expects to $ expand against the submit object, those expansions will expand to empty against a fresh submit object.  In practice, this can only affect the first line, so it would be ok to do the $() expansion here, but it might be better qargs an explict argument to _submit_itemdata rather than an implicit one.

Reply:
Good catch.

I'll look at how tricky that is, but at the moment, I'm leaning towards starting with a copy of the current submit object as the fix, instead.

Reply:
Looks like the Python-side approach is simpler to code; fix uploaded, with a test.